Microsoft.Diagnostics.NETCore.Client versioning prevents adopting preview for validation

The current versioning scheme of the preview versions of Microsoft.Diagnostics.NETCore.Client have the pattern of `0.2.0-preview.#####.#`. The release version has a pattern of `0.2.######`. This causes NuGet downgrade issues if a referenced library is using the shipped version and the project is attempting to adopt the preview version for validation.

For example, the .NET Monitor project runs into this warning (treated as an error) because Microsoft.Diagnostics.Tracing.TraceEvent references the shipped version:
```
src/Tests/Microsoft.Diagnostics.Monitoring.WebApi.UnitTests/Microsoft.Diagnostics.Monitoring.WebApi.UnitTests.csproj(0,0): error NU1605: (NETCORE_ENGINEERING_TELEMETRY=Restore) Warning As Error: Detected package downgrade: Microsoft.Diagnostics.NETCore.Client from 0.2.510501 to 0.2.0-preview.24576.1. Reference the package directly from the project to select a different version. 
 Microsoft.Diagnostics.Monitoring.WebApi.UnitTests -> Microsoft.Diagnostics.Monitoring.EventPipe 9.0.0-preview.24576.1 -> Microsoft.Diagnostics.Tracing.TraceEvent 3.1.16 -> Microsoft.Diagnostics.NETCore.Client (>= 0.2.510501) 
 Microsoft.Diagnostics.Monitoring.WebApi.UnitTests -> Microsoft.Diagnostics.Monitoring.EventPipe 9.0.0-preview.24576.1 -> Microsoft.Diagnostics.NETCore.Client (>= 0.2.0-preview.24576.1)
```

Newer preview releases need to have a higher version than the shipped version to allow adoption of the preview version for validation.

--- a/src/Microsoft.Diagnostics.NETCore.Client/Microsoft.Diagnostics.NETCore.Client.csproj
+++ b/src/Microsoft.Diagnostics.NETCore.Client/Microsoft.Diagnostics.NETCore.Client.csproj
@@ -14,6 +14,7 @@
     <IsShipping>true</IsShipping>
     <BuildingOutsideDiagnostics>false</BuildingOutsideDiagnostics>
     <BuildingOutsideDiagnostics Condition="'$(GitHubRepositoryName)' != 'diagnostics'">true</BuildingOutsideDiagnostics>
+    <PreReleaseVersionLabel />
   </PropertyGroup>
 
   <PropertyGroup Condition="$(BuildingOutsideDiagnostics)">

